File: backend/src/main/java/com/jbconnect/healthclinix/security/facade/SecurityFacadeImpl.java
package com.jbconnect.healthclinix.security.facade;

import org.springframework.stereotype.Component;

import com.jbconnect.healthclinix.entity.user.dto.UserRequestDTO;
import com.jbconnect.healthclinix.security.port.LoginPort;

@Component
public class SecurityFacadeImpl implements SecurityFacade {

	private final LoginPort<UserRequestDTO, String> loginPort;

	public SecurityFacadeImpl(LoginPort<UserRequestDTO, String> loginPort) {
		this.loginPort = loginPort;
	}

	@Override
	public String login(UserRequestDTO userRequestDTO) {
		return this.loginPort.execute(userRequestDTO);
	}

}

File: backendkotlin/src/main/kotlin/com/jbconnect/healthclinix/healthclinix/security/facade/SecurityFacadeImpl.kt
package com.jbconnect.healthclinix.healthclinix.security.facade

import com.jbconnect.healthclinix.healthclinix.entity.user.dto.UserRequestDTO
import com.jbconnect.healthclinix.healthclinix.security.port.LoginPort
import org.springframework.stereotype.Component


@Component
class SecurityFacadeImpl(val loginPort : LoginPort<UserRequestDTO, String>) : SecurityFacade {
    override fun login(userRequestDTO: UserRequestDTO): String {
        return this.loginPort.execute(userRequestDTO)
    }

}